rust/src/data_structures.rs

```diff
@@ -826,6 +1001,30 @@ pub struct DtlsParameters {
     pub fingerprints: Vec<DtlsFingerprint>,
 }
 
+impl DtlsParameters {
+    pub(crate) fn to_fbs(&self) -> web_rtc_transport::DtlsParameters {
+        web_rtc_transport::DtlsParameters {
+            role: Some(self.role.to_fbs()),
```

Comment:
Why is `role` optional in flatbuffers?

Reply:
AFAIR we allow it to be optional so we decide in worker side.

Reply:
Not sure what benefit it has because `None` case now has to be handled by both worker and library (we have `unwrap_or` below). It would be easier for everyone if it wasn't optional in the first place.

Reply:
I may be wrong but role None is used to indicate that this->role in DtlsTransport.cpp is not yet set (None is its default initial value). However role is optional in arguments given to webRtcTransport.connect() method. Those are different things.

Reply:
IMHO it makes sense that the **DtlsParameters** type has non optional DtlsRole, and having it defaulted to 'auto'.